src/analysis/calculus/times_cont_diff.lean

```diff
@@ -1626,6 +1626,22 @@ lemma times_cont_diff_within_at_snd {s : set (E × F)} {p : E × F} {n : with_to
   times_cont_diff_within_at 𝕜 n (prod.snd : E × F → F) s p :=
 times_cont_diff_snd.times_cont_diff_within_at
 
+/--
+The natural equivalence `(E × F) × G ≃ E × (F × G)` is smooth.
+-/
+lemma times_cont_diff_prod_assoc :
+  times_cont_diff 𝕜 ⊤ $ equiv.prod_assoc E F G :=
+(linear_isometry_equiv.coe_prod_assoc 𝕜 E F G) ▸
+(linear_isometry_equiv.prod_assoc 𝕜 E F G).times_cont_diff
```

Comment:
Do this and the next proof fail if you remove the first line (`(linear_isometry_equiv.coe_prod_assoc 𝕜 E F G) ▸`)(you're only rewriting with reflexivity anyway).

Reply:
Good point, I've dropped the redundant `rfl` rewrites.